supplier_server_mock/supplier_server/app.py

```diff
@@ -170,7 +170,26 @@ def booking():
 @app.route('/v1/booking/<booking_id>', methods=['DELETE'])
 @authorization_header
 def cancel_booking(booking_id):
-    return ('', 204)
+    try:
+        booked_at, product_id = utils.decode_booking_data(booking_id)
+    except binascii.Error:
+        raise exceptions.BadRequest(1004, 'Missing booking', 'Required argument \'booking_id\' was not found')
+    product = [p for p in constants.PRODUCTS if p["id"]== product_id][0]
+    # if we want to test that we need to store the cancelled booking id somewhere
+    if booking_id in product["cancelled_bookings"]:
+        raise exceptions.BadRequest(3003, 'Already cancelled', f'The booking with ID {booking_id} was already cancelled')
+    if not product["is_refundable"]:
+        raise exceptions.BadRequest(3004, 'Cancellation not possible', 'The booking cannot be cancelled, the product does not allow cancellations')    
+    booking_time = datetime.fromisoformat(booked_at)
```

Comment:
We need to check the date that was booked, not the date when the booking was created. We need to add that date to the `reservation_id` and then pass it to the `booking_id`.